Gems/ScriptCanvas/Code/Editor/Framework/Interpreter.cpp

```diff
@@ -0,0 +1,269 @@
+/*
+ * Copyright (c) Contributors to the Open 3D Engine Project.
+ * For complete copyright and license terms please see the LICENSE at the root of this distribution.
+ *
+ * SPDX-License-Identifier: Apache-2.0 OR MIT
+ *
+ */
+
+#include <Editor/Framework/Interpreter.h>
+
+#include <AzCore/Component/TickBus.h>
+#include <AzCore/Serialization/EditContext.h>
+#include <AzCore/Serialization/SerializeContext.h>
+#include <Builder/ScriptCanvasBuilder.h>
+#include <ScriptCanvas/Bus/ScriptCanvasBus.h>
+
+namespace ScriptCanvasEditor
+{
+    using namespace ScriptCanvas;
+    using namespace ScriptCanvasBuilder;
+
+    Interpreter::Interpreter()
```

Comment:
The Interpreter owns a Configuration, allowing the programmer/user to select a graph, and a Executor, which will run the compiled graph asset if possible. It provides status updates to the programmer/user on the status of both source and asset. It runs graphs (and stops them if needed, or desired) on the system tick thread.